Create Initial Cassandra Chat Table

car12lin12 opened the pull request:
The main cassandra chat table is created where the from_student_id and to_student_id & sent_timestamp form the primary key. This should be useful and efficient when getting messages for a chat between two users within a time range. The schema is found in repository/cassandra_chat_schema.cql

The cassandra database is initialized by executing the docker-compose.yaml. There is another branch called cassandra-example that shows examples of how to use the cassandra driver library; gocql, that is being used in this project. Further work in future PRs are required to integrate socket programming that will use the cassandra chat table.

car12lin12 committed:
example gocql select stmt using from and to student id

car12lin12 committed:
list all from & to example

car12lin12 committed:
just start db server

car12lin12 committed:
include sonar project key

car12lin12 committed:
omit sonar build for now bc only run locally

car12lin12 committed:
nothing to test yet

nuzurie committed:
STUD-167, STUD-168: create participants and rooms tables and modify messages table
Co-authored-by: amine <amine.bechrifa@gmail.com>
Co-authored-by: Mark Said <marksaid206@gmail.com>

nuzurie commented on domain/chat.go:
```diff
@@ -0,0 +1,18 @@
+	MessageBody string
+	TeamID string
+	SentTimestamp time.Time
+}
```
This needs to be changed to mirror the tables. Look for table discussion below

nuzurie commented on repository/cassandra_chat_schema.cql:
```diff
@@ -0,0 +1,13 @@
+    team_id         text,
+    sent_timestamp  timestamp,
+    PRIMARY KEY ( (from_student_id, to_student_id), sent_timestamp )
+) WITH CLUSTERING ORDER BY (sent_timestamp ASC);
```
1. What is the reason for having message_id as UUID if we don't use it. It's not a part of the primary key.
2. We could generalize to be a message for room, group or individual. For that purpose, we can remove to_student_id, and replace it with a room_id.
3. team_id can be replaced with the room_id.
4. room_id can be set as the partition key and sent_timestamp as the clustering key.
5. I would order by sent_timestamp DESC instead of ASC. This would allow you to get n newest messages instead of the first n in a chat.

car12lin12 replied:
good point, the message_id can be removed. I like the room_id idea! That'll be useful for retrieving messages in a group, we should have another table for a chat room's list of participants

nuzurie replied:
I made an attempt to address this, since this is time sensitive. Can you review the PR please.

nuzurie committed:
Create domains and define their interfaces

chat room, rooms for students, message and student domains and their interfaces defined

nuzurie committed:
STUD-128, STUD-129/domain, messaging: Create a chat app that connects with sockets, and saves the sent messages in the repository. Any sent messages are broadcasted to all online connections.

aminebechrifa committed:
Repository layer for chatrooms and students rooms

aminebechrifa committed:
Refractoring

aminebechrifa committed:
usecase layer chatrooms

aminebechrifa committed:
removing the exit after an error

removing the log.fatal

aminebechrifa committed:
indent

car12lin12 committed:
Merge pull request #6 from Airbenders-490/feature/STUD170-STUD169

Feature/stud170 stud169
